resource/fingerprint_test.go

```diff
@@ -56,10 +57,27 @@ func (s *FingerprintSuite) TestNewFingerprintTooBig(c *gc.C) {
 	c.Check(err, gc.ErrorMatches, `.*too big.*`)
 }
 
+func (s *FingerprintSuite) TestParseFingerprintOkay(c *gc.C) {
+	_, expected := newFingerprint(c, "spamspamspam")
+
+	fp, err := resource.ParseFingerprint(expected)
+	c.Assert(err, jc.ErrorIsNil)
+	hex := fp.String()
+
+	c.Check(hex, jc.DeepEquals, expected)
+}
+
+func (s *FingerprintSuite) TestParseFingerprintNonHex(c *gc.C) {
+	_, err := resource.ParseFingerprint("XYZ") // not hex
+
+	c.Check(err, gc.ErrorMatches, `.*odd length hex string.*`)
+}
+
 func (s *FingerprintSuite) TestGenerateFingerprint(c *gc.C) {
 	expected, _ := newFingerprint(c, "spamspamspam")
+	data := bytes.NewBufferString("spamspamspam")
```

Comment:
FYI, strings.NewReader is more appropriate here, it's the string->io.Reader that you're looking for.